xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1056 @@
+        printer.print(
+            "(",
+            format_hex(self.packet_type.value.data),
+            ", ",
+            format_hex(self.packet_id.value.data),
+            ")",
+        )
+
+
+# TODO: add successor basic blocks
+@irdl_op_definition
+class DMAStartOp(IRDLOperation):
+    name = "AIE.dmaStart"
+    channelDir: IntegerAttr[Annotated[IntegerType, i32]] = attr_def(
+        IntegerAttr[Annotated[IntegerType, i32]]
+    )
+    channelIndex: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    dest: Block = successor_def()
+    chain: Block = successor_def()
+    # result: OpResult	= result_def(i1)
+
+    traits = frozenset([IsTerminator()])
+
+    def __init__(
+        self,
+        channelDir: IntegerAttr[i32],
+        channelIndex: IntegerAttr[i32],
+        dest: Block,
+        chain: Block,
+    ):
+        super().__init__(
+            attributes={"channelDir": channelDir, "channelIndex": channelIndex},
+            successors=[dest, chain],
+            # result_types=[i1]
+        )
+
+    def print(self, printer: Printer):
+        direction = "MM2S" if self.channelDir.value.data == 0 else "S2MM"
+        printer.print("(", direction, ", ", self.channelIndex.value.data, ", ")
+        printer.print_block_name(self.dest)
+        printer.print(", ")
+        printer.print_block_name(self.chain)
+        printer.print(")")
+
+
+@irdl_op_definition
+class DebugOp(IRDLOperation):
+    name = "AIE.debug"
+    arg: Operand = operand_def(AnyAttr())
+
+    def __init__(self, arg: AnyAttr()):
+        super().__init__(operands=[arg])
+
+
+@irdl_op_definition
+class DeviceOp(IRDLOperation):
+    name = "AIE.device"
+
+    region: Region = region_def()
+
+    device: IntegerAttr[Annotated[IntegerType, i32]] = attr_def(
+        IntegerAttr[Annotated[IntegerType, i32]]
+    )
+    traits = frozenset([SymbolTable()])
+
+    def __init__(self, device: IntegerAttr[i32], region: Region):
+        super().__init__(attributes={"device": device}, regions=[region])
+
+    def print(self, printer: Printer):
+        printer.print("(")
+        device_str = "xcvc1902" if self.device.value.data == 0 else ""
+        printer.print(device_str)
+        printer.print(") ")
+        printer.print_region(self.region)
+
+
+@irdl_op_definition
+class ExternalBufferOp(IRDLOperation):
+    name = "AIE.external_buffer"
+
+    sym_name: StringAttr = attr_def(StringAttr)
+    buffer: OpResult = result_def(memref.MemRefType)
+
+    def __init__(
+        self,
+        sym_name: str,
+        shape: ArrayAttr[AnyIntegerAttr],
+        element_type: _MemRefTypeElement,
+    ):
+        super().__init__(
+            attributes={"sym_name": StringAttr(sym_name)},
+            result_types=[
+                memref.MemRefType.from_element_type_and_shape(element_type, shape)
+            ],
+        )
+
+
+@irdl_op_definition
+class FlowOp(IRDLOperation):
+    name = "AIE.flow"
+
+    sourceBundle: WireBundleAttr = attr_def(WireBundleAttr)
```

Comment:
similarly here for attribute names and python vs C++ conventions, I'd recommend making python pretty and adding the translation hook thing. I'm not sure if everyone else will agree with this though, definitely a matter of personal preference

Reply:
What would be your suggestion for the name of this attribute?